neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -201,7 +201,8 @@ private StackItem GetRegisteredValidators(ApplicationEngine engine, VMArray args
 
         public IEnumerable<(ECPoint PublicKey, BigInteger Votes)> GetRegisteredValidators(Snapshot snapshot)
         {
-            return snapshot.Storages.Find(new[] { Prefix_Validator }).Select(p =>
+            byte[] prefix_key = StorageKey.CreateSearchPrefix(Hash, new[] { Prefix_Validator });
```

Comment:
It looks clear, but maybe add a comment about the `Hash` of the validators being the the one of `NativeContract`.
This `Hash` gives the idea of the local hash of the NeoToken.